readthedocs/proxito/views/mixins.py

```diff
@@ -210,7 +209,15 @@ def system_redirect(self, request, final_project, lang_slug, version_slug, filen
         resp['X-RTD-Redirect'] = 'system'
         return resp
 
-    def canonical_redirect(self, request, final_project, version_slug, filename):
+    def canonical_redirect(
+        self,
+        request,
+        final_project,
+        version_slug,
+        filename,
+        redirect_type=None,
```

Comment:
What are the possible values here? We should document them, and maybe make usage of enums as well in case it makes sense.

Reply:
we have constants for these already https://github.com/readthedocs/readthedocs.org/blob/dont-depend-on-injected-fields/readthedocs/proxito/constants.py. But would love to change them to enums some day :)

Reply:
I'd say that you should create a refactor issue/project listing all these things we want to upgrade. It will be easier to prioritize this work if we have a place where we are tracking it. Maybe is a good project for the low December week? 🤔